persistent/Database/Persist/Quasi/Internal.hs

```diff
@@ -984,10 +985,16 @@ takeConstraint ps entityName defs (n :| rest) =
                     Just $ pure (takeForeign ps entityName rest)
                 }
         "Primary" ->
-            mempty
-                { entityConstraintDefsPrimaryComposite =
-                    SetOnce (takeComposite (unboundFieldNameHS <$> defs) rest)
-                }
+            let
+                unboundComposite =
+                    takeComposite (unboundFieldNameHS <$> defs) rest
+            in
+                mempty
+                    { entityConstraintDefsPrimaryComposite =
+                        SetOnce unboundComposite
+                    , entityConstraintDefsUniques =
+                        Just $ pure $ compositeToUniqueDef entityName defs unboundComposite
```

Comment:
This is the real change. We're adding a `UniqueDef` based on the composite primary key.